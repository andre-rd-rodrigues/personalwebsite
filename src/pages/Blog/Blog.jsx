import React from "react";
import PageContainer from "components/PageContainer/PageContainer";
import { articles } from "mocks/articles";
import ReactGA from "react-ga4";
import ArticlesGrid from "./ArticlesGrid/ArticlesGrid";
import styles from "./blog.module.scss";

const Blog = () => {
  return (
    <PageContainer bgColor="light-grey" className={styles.container}>
      <div className={styles.title}>
        <div className={styles.blogImage}></div>
        <h1>Curiosity Gem</h1>
        <p>Get the real experience and visit my official blog website</p>
        <button
          onClick={() =>
            ReactGA.event({
              category: "Navigation",
              action: "click_navigation_to_blog"
            })
          }
        >
          <a
            target="_blank"
            rel="noreferrer"
            href="https://blog.andrerodrigo.com"
          >
            get started
          </a>
        </button>
      </div>
      <div style={styles.articles}>
        <ArticlesGrid articles={articles} />
      </div>
    </PageContainer>
  );
};

export default Blog;
