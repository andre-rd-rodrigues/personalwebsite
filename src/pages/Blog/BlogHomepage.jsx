import ArticlesGrid from "components/Blog/ArticlesGrid/ArticlesGrid";
import TopArticle from "components/Blog/TopArticle/TopArticle";
import { blog } from "mocks/data";
import React from "react";
import styles from "./blog.module.scss";

function BlogHomepage() {
  return (
    <div className={styles.blogHomepage}>
      <h1>ideas & thoughts</h1>
      <TopArticle article={blog.articles[0]} />
      <h2 className="recent_post">Recent posts</h2>
      <ArticlesGrid articles={blog.articles} />
    </div>
  );
}

export default BlogHomepage;
