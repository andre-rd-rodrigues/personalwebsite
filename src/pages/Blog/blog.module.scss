@import "assets/styles/mixins";

.container {
  padding-top: 0;
  min-height: 97vh;
}

.title {
  display: flex;
  align-items: center;
  justify-content: center;
  min-height: 80vh;
  flex-direction: column;
  color: $dark;

  h1 {
    font-size: var(--fs-xl);
  }

  button {
    margin-top: 20px;
    @include button($dark);
    background-color: $dark;
    color: white;
    &:hover {
      opacity: 0.8;
    }
  }
}

.blogImage {
  background: url("https://images.unsplash.com/photo-1597589827703-f4b68eafa0ce?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1544&q=80")
    center center no-repeat;
  background-size: cover !important;
  width: 90px;
  height: 90px;
  border-radius: 50%;
}
