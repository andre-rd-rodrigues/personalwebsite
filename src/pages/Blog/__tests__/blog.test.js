import { screen } from "@testing-library/react";
import { render } from "utils/tests/test-utils";

import Blog from "../Blog";

describe("Blog", () => {
  it("renders page correctly", () => {
    const { asFragment } = render(<Blog />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders Blog title correctly", () => {
    render(<Blog />);

    const titleElement = screen.getByText(/Curiosity Gem/i);

    expect(titleElement).toBeInTheDocument();
  });

  it("redirects to blog website on link click", () => {
    render(<Blog />);

    const linkElement = screen.getByText(/Go to blog/i);

    expect(linkElement).toHaveAttribute("href", "https://www.curiositygem.com");
  });
});
