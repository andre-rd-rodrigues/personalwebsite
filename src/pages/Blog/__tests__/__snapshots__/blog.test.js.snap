// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog renders page correctly 1`] = `
<DocumentFragment>
  <div
    class="container container"
  >
    <div
      class="wrapper"
      style="opacity: 0;"
    >
      <div
        class="title"
      >
        <div
          class="blogImage"
        />
        <h1>
          Curiosity Gem
        </h1>
        <p>
          Get the real experience and visit my official blog website
        </p>
        <button>
          <a
            href="https://www.curiositygem.com"
            rel="noreferrer"
            target="_blank"
          >
            go to blog
          </a>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
